### DISCUSSION TITLE
What is the redirect URL for OAuth?

### DISCUSSION THREAD

**Discussion by emregency** [directus/directus#3678] [Archive]
Answer: <directus_public_url>/auth/oauth/<provider_name>/callback

**Comment by benhaynes** [comment]
Answering your own question?? 😄

**Reply by emregency** [reply]
Yeah sorry for that 🤣. I thought someone could also encounter the same issue.

**Reply by benhaynes** [reply]
I like it! Next time I think you can answer separately so that we can mark your response as the answer! 👍

**Reply by mscbpi** [reply]
THANK YOU :) useful

**Reply by dijs** [reply]
Where is this set in directus? I am getting: ```"/auth/oauth/github/callback?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch```

**Reply by joselcvarela** [reply]
Hello @dijs 
Have you defined your env variables for oAuth? Here you can check how it's done:
https://docs.directus.io/guides/api-config/#oauth-single-sign-on-sso-openid

Don't forget to set your `PUBLIC_URL` as well. For example, if your server is at: `https://example.com` so your PUBLIC_URL will be the same `https://example.com`

**Comment by mykhailo-sindieiev** [comment]
I am tuning auth0 **openid** sso integration and the redirect url is `https://<domain_name>/auth/login/auth0/callback`
